bps/org.wso2.developerstudio.eclipse.artifact.bpel/src/org/wso2/developerstudio/eclipse/artifact/bpel/ui/wizard/RegistryResourceBrowserDialog.java

```diff
@@ -272,19 +272,16 @@ private void setSelectedPath(String selectedPath) {
 
 	@SuppressWarnings("unchecked")
 	private void openRegistryResourceProviderDialog() {
-		 hide();
-		try {
-			RegistryResourseFileProviderDialog registryResourceProviderSelector = new RegistryResourseFileProviderDialog(
-					getParentShell(), new Class[] { IRegistryFile.class }, new HashMap<String, List<String>>());
-			registryResourceProviderSelector.create();
-			registryResourceProviderSelector.getShell().setText("Workspace Element Providers");
-			registryResourceProviderSelector.open();
-			if (registryResourceProviderSelector.getReturnCode() == Window.OK) {
-				setSelectedPath(registryResourceProviderSelector.getSelectedPath());
-			}
-		} finally {
-			show();
+		hide();
+		RegistryResourseFileProviderDialog registryResourceProviderSelector = new RegistryResourseFileProviderDialog(
+				getParentShell(), new Class[] { IRegistryFile.class }, new HashMap<String, List<String>>());
+		registryResourceProviderSelector.create();
+		registryResourceProviderSelector.getShell().setText("Workspace Element Providers");
+		registryResourceProviderSelector.open();
+		if (registryResourceProviderSelector.getReturnCode() == Window.OK) {
```

Comment:
registryResourceProviderSelector.getReturnCode() and Window.OK are int values. Thanks for the feedback.